cc/crypto/hpke/recipient_context.cc

```diff
@@ -16,7 +16,180 @@
 
 #include "cc/crypto/hpke/recipient_context.h"
 
+#include "absl/status/statusor.h"
+#include "absl/strings/string_view.h"
+#include "openssl/hpke.h"
+
 namespace oak::crypto {
+namespace {
+// Validates that the public and private key pairing is valid for HPKE. If the public and private
+// keys are valid, the recipient_keys argument will be an initialized HPKE_KEY.
+absl::Status ValidateKeys(std::vector<uint8_t>& public_key_bytes,
```

Comment:
Should keys be validated for the sender context as well?
If so, this function could be moved to `utils.cc`